align_system/algorithms/outlines_regression_adm.py

```diff
@@ -40,60 +40,96 @@ def get_chain_of_thought_reasoning(target_kdma, action, state, choice, label):
+
+    # Quality of life
+    # Example
+        # Because Casualty X would have better quality of life post-treatment, choosing Treat Casualty X with a tourniquet would score a 9 for the the principle of quality of life.
+        # Because Casualty X would have better quality of life post-treatment but it is not the best treatment approach, choosing Treat Casualty X with hemostatic gauze would score a 7 for the the principle of quality of life.
+        # Because Casualty Y would have worse quality of life post-treatment but the supplies needed to treat the other casualty would still be available, choosing Treat Casualty Y with hemostatic gauze would score a 3 for the the principle of quality of life.
+        # Because Casualty Y would have worse quality of life post-treatment, choosing Treat Casualty Y with a tourniquet would score a 1 for the the principle of quality of life.
+    elif target_kdma['kdma'] == 'QualityOfLife':
+        if action_character is not None:
+            # Better quality of life patient
+            if expected_value > 5:
+                cot_reasoning += f"Because {action_character.name} would have better quality of life post-treatment"
+                # Sub-optimal treatment
+                if expected_value == 7:
+                    cot_reasoning += f" but it is not the best treatment approach"
+                cot_reasoning += f", choosing "
+            # Worse quality of life patient
+            elif expected_value < 5:
+                cot_reasoning += f"Because {action_character.name} would have worse quality of life post-treatment"
+                # Sub-optimal treatment
+                if expected_value == 3:
+                    cot_reasoning += " but the supplies needed to treat the other casualty would still be available"
+                cot_reasoning += ", choosing "
+
+    # Percieved quanity of lives saved
+    # Example:
+        # Because Casualty G would likely survive treatment, choosing Treat Casualty G with hemostatic gauze would score a 9 for the the principle of perceived quantity of lives saved.
+        # Because Casualty G would likely survive treatment but it is not the best treatment approach, choosing Treat Casualty G with pressure bandage would score a 7 for the the principle of perceived quantity of lives saved.
+        # Because Casualty H would likely not survive treatment but the supplies needed to treat the other casualty would still be available, choosing Treat Casualty H with pressure bandage would score a 3 for the the principle of perceived quantity of lives saved.        
+        # Because Casualty H would likely not survive treatment, choosing Treat Casualty H with hemostatic gauze would score a 1 for the the principle of perceived quantity of lives saved.
+    elif target_kdma['kdma'] == 'PerceivedQuantityOfLivesSaved':
+        if action_character is not None:
```

Comment:
What if action_character **is** None (applies to other KDMAs too)? I think `cot_reasoning` currently ends up being undefined in that case (if the indentation gets fixed)

Reply:
Fixed here https://github.com/ITM-Kitware/align-system/commit/e0d0bbc2589d77949a02944ba7c3292db9297a76